### PR TITLE
OCP4 STIG: Several RBAC related controls are no longer Inherently Met, let's use a manual rule to cut duplication

### PR DESCRIPTION
**Pull request by jhrozek**
#### Description:

This patch reflects the changes to the OCP4 STIG draft. Several rules
that used to be marked as Inherently Met with justification pointing to
RBAC are now applicable/configurable with a manual rule that asks for
the RBAC rules to be reviewed.

In order to cut down on duplication and provide the same rendered
spreadsheet output as DISA expects, I also implemented a manual rule
that covers the requirement and its text can be resued.

#### Rationale:

OCP4 STIG

#### Review Hints:

- I will post a link to the SRG export later

### CODE REVIEW

**Comment by jhrozek** [comment]
@Vincent056 @rhmdnd PTAL.
To review, please compare the [latest STIG draft](https://docs.google.com/spreadsheets/d/1vRjx7fi6lg2qLstNJWY08qCm4s-slyKUrfrjj9LCRAQ/edit#gid=312154131) with the [SRG export](https://github.com/ComplianceAsCode/content/suites/13083663137/artifacts/710061820)

**Comment by jhrozek** [comment]
CC @xiaojiey - this is a new manual rule so I guess the only thing to test here would be if the steps make sense. There is no automated check or remediation.

**Comment by BhargaviGudi** [comment]
Verification passed with 4.14.0-0.nightly-2023-05-23-103225
<pre>
1. Deployed compliance operator with ComplianceAsCode/compliance-operator
2. Deployed content with PR #10612 
3. Created ssb with stig profiles
$ oc compliance bind -N test -S default profile/upstream-ocp4-stig profile/upstream-ocp4-stig-node
Creating ScanSettingBinding test
$ oc get ssb
NAME   AGE
test   3s
$ oc get suite -w
NAME   PHASE       RESULT
test   LAUNCHING   NOT-AVAILABLE
test   LAUNCHING   NOT-AVAILABLE
test   LAUNCHING   NOT-AVAILABLE
test   RUNNING     NOT-AVAILABLE
test   RUNNING     NOT-AVAILABLE
test   RUNNING     NOT-AVAILABLE
test   AGGREGATING   NOT-AVAILABLE
test   AGGREGATING   NOT-AVAILABLE
test   AGGREGATING   NOT-AVAILABLE
test   DONE          NON-COMPLIANT
test   DONE          NON-COMPLIANT
4. Check for rbac-least-privilege
$ oc get ccr | grep rbac-least-privilege
upstream-ocp4-stig-rbac-least-privilege                                                         MANUAL   high

$ oc describe ccr upstream-ocp4-stig-rbac-least-privilege
Name:         upstream-ocp4-stig-rbac-least-privilege
Namespace:    openshift-compliance
Labels:       compliance.openshift.io/check-severity=high
              compliance.openshift.io/check-status=MANUAL
              compliance.openshift.io/scan-name=upstream-ocp4-stig
              compliance.openshift.io/suite=test
Annotations:  compliance.openshift.io/rule: rbac-least-privilege
API Version:  compliance.openshift.io/v1alpha1
Description:  Ensure that the RBAC setup follows the principle of least privilege
Role-based access control (RBAC) objects determine whether a user is allowed to perform a given action within a project. If users or groups exist that are bound to roles they must not have, modify the user or group permissions using the following cluster and local role binding commands: Remove a User from a Cluster RBAC role by executing the following: oc adm policy remove-cluster-role-from-user role username Remove a Group from a Cluster RBAC role by executing the following: oc adm policy remove-cluster-role-from-group role groupname Remove a User from a Local RBAC role by executing the following: oc adm policy remove-role-from-user role username Remove a Group from a Local RBAC role by executing the following: oc adm policy remove-role-from-group role groupname NOTE: For additional information. https://docs.openshift.com/container-platform/latest/authentication/using-rbac.html
Id:            xccdf_org.ssgproject.content_rule_rbac_least_privilege
Instructions:  The administrator must verify that Openshift is configured with the necessary RBAC access controls.

Review the RBAC configuration.

As the cluster-admin, view the cluster roles and their associated rule sets by executing the following::

oc describe clusterrole.rbac

Now view the current set of cluster role bindings, which shows the users and groups that are bound to various roles by executing the following:

oc describe clusterrolebinding.rbac

Local roles and bindings can be determined using the follow commands by executing the following:

oc describe rolebinding.rbac

If these results show users with privileged access that do not require that access, this is a finding.
Kind:  ComplianceCheckResult
Metadata:
  Creation Timestamp:  2023-05-25T07:29:40Z
  Generation:          1
  Managed Fields:
    API Version:  compliance.openshift.io/v1alpha1
    Fields Type:  FieldsV1
    fieldsV1:
      f:description:
      f:id:
      f:instructions:
      f:metadata:
        f:annotations:
          .:
          f:compliance.openshift.io/rule:
        f:labels:
          .:
          f:compliance.openshift.io/check-severity:
          f:compliance.openshift.io/check-status:
          f:compliance.openshift.io/scan-name:
          f:compliance.openshift.io/suite:
        f:ownerReferences:
          .:
          k:{"uid":"a095989c-df68-44b0-8f21-ab55aaba3f90"}:
      f:rationale:
      f:severity:
      f:status:
    Manager:    compliance-operator
    Operation:  Update
    Time:       2023-05-25T07:29:40Z
  Owner References:
    API Version:           compliance.openshift.io/v1alpha1
    Block Owner Deletion:  true
    Controller:            true
    Kind:                  ComplianceScan
    Name:                  upstream-ocp4-stig
    UID:                   a095989c-df68-44b0-8f21-ab55aaba3f90
  Resource Version:        70655
  UID:                     c0d11aae-cb3b-4540-96c5-715e343ddaed
Rationale:                 Controlling and limiting users access to system services and resources is key to securing the platform and limiting the intentional or unintentional comprimising of the system and its services. OpenShift provides a robust RBAC policy system that allows for authorization policies to be as detailed as needed. Additionally there are two layers of RBAC policies, the first is Cluster RBAC policies which administrators can control who has what access to cluster level services. The other is Local RBAC policies, which allow project developers/administrators to control what level of access users have to a given project or namespace.
Severity:                  high
Status:                    MANUAL
Events:                    <none>
</pre>

File: shared/macros/10-ocil.jinja [hunk 1]
```diff
@@ -1248,3 +1248,29 @@ Review the list of users and groups who have {{{ verb }}} access to the cluster
...
+
+#}}
+{{%- macro ocil_rbac_least_privilege() -%}}
+The administrator must verify that Openshift is configured with the necessary RBAC access controls.
+
+Review the RBAC configuration.
+
+As the cluster-admin, view the cluster roles and their associated rule sets by executing the following::
+
+oc describe clusterrole.rbac
```
Comment (rhmdnd) [inline shared/macros/10-ocil.jinja:1264]: Holy output...

**Review by rhmdnd — APPROVED** [review]
/lgtm

Thanks for drying things up.

**Pull request merged**